lib/engine-addon.js

```diff
@@ -122,19 +122,18 @@ function buildVendorCSSTree(vendorTree) {
   });
 }
 
-function buildEngineAppTree(engineSourceTree) {
-  if (!engineSourceTree) {
-    var treePath = path.resolve(this.root, this.treePaths['addon']);
-    if (existsSync(treePath)) {
-      engineSourceTree = this.treeGenerator(treePath);
-    }
+function buildEngineAppTree() {
+  var engineSourceTree;
+  var treePath = path.resolve(this.root, this.treePaths['addon']);
```

Comment:
In a future refactor we should migrate to `this.treePathFor('addon')` if it is present (it will be present on 2.12+)